x-pack/plugins/maps/public/classes/layers/layer.tsx

```diff
@@ -423,7 +423,7 @@ export class AbstractLayer implements ILayer {
 
   renderSourceSettingsEditor({ onChange }: SourceEditorArgs) {
     const source = this.getSourceForEditing();
-    return source.renderSourceSettingsEditor({ onChange });
+    return source.renderSourceSettingsEditor({ onChange, currentLayerType: this._descriptor.type });
```

Comment:
Can you change ISource.renderSourceSettingsEditor definition to include currentLayerType. Its currently `renderSourceSettingsEditor({ onChange }: SourceEditorArgs): ReactElement<any> | null;`, should be `renderSourceSettingsEditor(sourceEditorArgs: SourceEditorArgs): ReactElement<any> | null;`